identity_stardust/src/network/network_name.rs

```diff
@@ -0,0 +1,124 @@
+// Copyright 2020-2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use std::borrow::Cow;
+
+use core::convert::TryFrom;
+use core::fmt::Display;
+use core::fmt::Formatter;
+use core::ops::Deref;
+use std::fmt::Debug;
+
+use serde::Deserialize;
+use serde::Serialize;
+
+use crate::error::Error;
+use crate::error::Result;
+
+/// Network name compliant with the [`crate::StardustDID`] method specification.
+#[derive(Clone, Hash, PartialEq, Eq, PartialOrd, Ord, Deserialize, Serialize)]
+#[repr(transparent)]
+pub struct NetworkName(pub(crate) Cow<'static, str>);
```

Comment:
This should not be `pub(crate)`, that allows bypassing its invariants elsewhere in the code.

```suggestion
pub struct NetworkName(Cow<'static, str>);
```

Reply:
If it is not accessible in the `stardust_did` module it will not be possible to construct `const StardustDID::DEFAULT_NETWORK : NetworkName` (unless we reintroduce the `const fn main_network() -> Self` function).

Reply:
We can still expose static instances for well-known networks like `dev` and `main`, possibly using `oncecell` if necessary.